File: DatePickerView/PickerView/ViewController.m
//
//  ViewController.m
//  PickerView
//
//  Created by Jacky on 2016/12/16.
//  Copyright © 2016年 jacky. All rights reserved.
//

#import "ViewController.h"
#import "JCalendar.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    JCalendar *calendat = [[JCalendar alloc]initWithFrame:CGRectMake(0, 100, [UIScreen mainScreen].bounds.size.width, 300)];
    calendat.backgroundColor = [UIColor redColor];
    [self.view addSubview:calendat];
    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: FirstWidget/FirstWidget/FirstWidget/ViewController.swift
//
//  ViewController.swift
//  FirstWidget
//
//  Created by Jacky on 16/9/9.
//  Copyright © 2016年 jacky. All rights reserved.
//

import UIKit

class ViewController: UIViewController,UITableViewDelegate,UITableViewDataSource {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        self.view.backgroundColor = UIColor.redColor()
        
        self.view .addSubview(tableView); 
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

//MARK: tableview datasource
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 4;
    }
    
    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
        return 40;
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        return UITableViewCell.init(frame: CGRectMake(0, 0, 90, 90));
    }
//MARK: tableview delelgate
  
    //MARK: lazyl load
    
    var tableView:UITableView{
        let tmp = UITableView.init(frame:UIScreen.mainScreen().bounds)
        tmp.backgroundColor = UIColor.yellowColor()
        
        return tmp;
        
    }

    
}




